impl/src/metadata.rs

```diff
@@ -37,7 +79,12 @@ const SYSTEM_CARGO_BIN_PATH: &str = "cargo";
  * <https://github.com/rust-lang/cargo/pull/5122>
  */
 pub trait MetadataFetcher {
-  fn fetch_metadata(&self, files: &CargoWorkspaceFiles) -> Result<Metadata>;
+  fn fetch_metadata(
+    &self,
+    files: &CargoWorkspaceFiles,
+    binary_dep_info: Option<&HashMap<String, cargo_toml::Dependency>>,
+    override_lockfile: Option<PathBuf>,
```

Comment:
Ignored this comment?

Reply:
I don't think that `override_lockfile` should go with `CargoWorkspaceFiles` since it's intended to be a unique lockfile only generated by `cargo-raze`. The reason this file exists in the first place was `(b)`. That information does not exist within the projects cargo metadata because of the inconsistency in how Cargo handles binary dependencies. Raze generates a new lockfile to make sure subsequent runs are reproducible.

I actually feel that the `CargoWorkspaceFiles` struct should no longer exist. https://github.com/google/cargo-raze/pull/274 introduced a way to cleanly embed Raze settings into cargo metadata and because of this I feel we should actually be using a [`MetadataCommand`](https://docs.rs/cargo_metadata/0.12.0/cargo_metadata/struct.MetadataCommand.html) to load settings. It should no longer matter what `Cargo.toml` and `Cargo.lock` file is being used, we can the metadata from anywhere within a cargo workspace and parse settings. In the context of supporting workspaces, this would allow users to keep `CrateSettings` in the relevant workspace member toml files and have them be joined in an intelligent way to be used for the rest of the `raze` command. Additionally, should we need to locate the cargo files, `workspace_root` should always be the directory where the cargo files live for currently supported use cases (single/root package projects).

Somewhat related to this issue, there's work in https://github.com/UebelAndre/cargo-raze/commit/332ad931f374ea28e3100fe46c92e99dd51b6ae2 which is inspired by conversations in https://github.com/google/cargo-raze/pull/222 but also starts to point out that the `CargoWorkspaceFiles` is not needed. But for now I feel strongly that `override_lockfile` represents something different than `CargoWorkspaceFiles`. Though I agree with your point. I'd like to either delete the need for `override_lockfile` and have everything live in normal Cargo files or delete `CargoWorkspaceFiles` and use Cargo Metadata to gather the necessary information

Reply:
¯\\\_(ツ)_/¯